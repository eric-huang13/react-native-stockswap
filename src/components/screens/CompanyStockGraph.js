import React, { Component } from "react";
import {
  Text,
  View,
  StyleSheet,
  Dimensions,
} from "react-native";
import { SlideAreaChart } from "react-native-slide-charts";

export default class CompanyStockGraph extends Component {
    render() {
          return (
            <View style={styles.container}>
              <SlideAreaChart
                data={this.props.graphData}
                yRange={this.props.range}
                width={Dimensions.get("window").width - 60}
                height={200}
                style={{ backgroundColor: "#f0f4fc" }}
                axisWidth={5}
                axisHeight={5}
                chartLineColor={"rgb(8, 177, 40)"}
                // chartPaddingTop={-47}
                throttleAndroid={true}
                fillColor={"#f0f4fc"}
                cursorProps={{
                  cursorColor: "#333",
                  cursorLine: true,
                  cursorColor: "black",
                  cursorMarkerWidth: 11,
                  cursorMarkerHeight: 15,
                  cursorBorderColor: "black",
                }}
                paddingBottom={10}
                paddingLeft={3}
                //Y AXIS
                yAxisProps={{
                  showBaseLine: false,
                  verticalLineWidth: 0,
                  horizontalLineWidth: 0,
                  axisLabelAlignment: "top",
                  rotateAxisLabel: true,
                  labelTopPadding: 0,
      
                  //Set graph Y numbers
                  numberOfTicks: 0,
                  markFirstLine: true,
                  hideMarkers:true,
                  // interval:12,
                }}
                //TOOLTIP
                toolTipProps={{
                  displayTriangle: false,
                  height: 50,
                  fontSize: 14,
                  borderRadius: 8,
                  backgroundColor: "#f0f4fc",
                  toolTipTextRenderers: [
                    ({ scaleX, x }) => ({
                      text: new Date(scaleX.invert(x)).toLocaleDateString("en-US", {
                        hour: "2-digit",
                        minute: "numeric",
                      }),
                    }),
                    ({ scaleY, y }) => ({
                      text: "$" + scaleY.invert(y).toFixed(2).toString(),
                    }),
                  ],
                }}
              />
            </View>
          );
        }
      }
      const styles = StyleSheet.create({
        container: {
          // margin: 12,
          // borderWidth: 1,
          flex: 1,
          // padding: 20,
          backgroundColor: "#f0f4fc",
        },
      });
      