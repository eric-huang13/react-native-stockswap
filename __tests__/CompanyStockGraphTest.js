import React from 'react';
import CompanyStockGraph from '../src/components/SearchTabComponents/CompanyStockGraph';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<CompanyStockGraph />);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});