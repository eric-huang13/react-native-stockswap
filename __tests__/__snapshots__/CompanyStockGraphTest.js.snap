// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flex": 1,
    }
  }
>
  <SlideAreaChart
    alwaysShowIndicator={true}
    animated={true}
    axisHeight={5}
    axisWidth={5}
    chartLineColor="#91f2b1"
    chartLineWidth={3}
    chartPaddingTop={16}
    cursorProps={
      Object {
        "cursorBorderColor": "black",
        "cursorColor": "black",
        "cursorLine": true,
        "cursorMarkerHeight": 15,
        "cursorMarkerWidth": 11,
      }
    }
    curveType={[Function]}
    data={Array []}
    height={200}
    paddingBottom={10}
    paddingLeft={3}
    paddingRight={8}
    paddingTop={8}
    renderFillGradient={[Function]}
    shouldCancelWhenOutside={true}
    style={
      Object {
        "backgroundColor": "#2a334a",
      }
    }
    throttleAndroid={true}
    toolTipProps={
      Object {
        "backgroundColor": "#8b64ff",
        "borderRadius": 18,
        "displayTriangle": false,
        "fontSize": 14,
        "height": 50,
        "textStyles": Array [
          Object {
            "color": "white",
          },
          Object {
            "color": "white",
            "fontSize": 14.1,
            "fontWeight": "700",
          },
        ],
        "toolTipTextRenderers": Array [
          [Function],
          [Function],
        ],
      }
    }
    width={710}
    xScale="linear"
    yAxisProps={
      Object {
        "axisLabelAlignment": "top",
        "hideMarkers": true,
        "horizontalLineWidth": 0,
        "labelTopPadding": 0,
        "markFirstLine": true,
        "numberOfTicks": 0,
        "rotateAxisLabel": true,
        "showBaseLine": false,
        "verticalLineWidth": 0,
      }
    }
  />
</ForwardRef(SafeAreaView)>
`;
